src/server/pps/server/api_server.go

```diff
@@ -1816,6 +1872,9 @@ func (a *apiServer) validatePipelineRequest(request *pps.CreatePipelineRequest)
 	if request.Spout != nil && request.Autoscaling {
 		return errors.Errorf("autoscaling can't be used with spouts (spouts aren't triggered externally)")
 	}
+	if !request.Autoscaling && request.MaximumExpectedUptime != nil {
```

Comment:
This works.  I think it's safe to rely on this precondition in the pipeline controller; i.e. don't check autoscaling, just check maximum_expected_uptime.

Now that I think about it, it would be nice to update workers_started_at regardless of whether the pipeline is autoscaling or maximum_expected_uptime is set.

Reply:
@jrockway so should I just remove this completely?

Reply:
yeah, I think if maximum expected uptime is set, only use that to determine whether or not to do the check.